tensorflow/core/common_runtime/dml/dml_heap_allocator.cc

```diff
@@ -17,40 +17,233 @@ limitations under the License.
 
 #include "dml_util.h"
 #include "tensorflow/core/lib/strings/numbers.h"
-
+#include "tensorflow/core/util/env_var.h"
 namespace tensorflow {
 
+static bool GetTilingEnabled(ID3D12Device* device) {
+  bool force_placed_resources = false;
+  const char* force_placed_resources_string =
+      std::getenv("TF_DIRECTML_FORCE_PLACED_RESOURCES");
+  if (force_placed_resources_string != nullptr) {
+    if (strcmp("false", force_placed_resources_string) == 0) {
+      force_placed_resources = false;
+    } else if (strcmp("true", force_placed_resources_string) == 0) {
+      force_placed_resources = true;
+    } else {
+      LOG(ERROR) << "The TF_DIRECTML_FORCE_PLACED_RESOURCES environment "
+                    "variable is set "
+                    "but could"
+                 << " not be parsed: \"" << force_placed_resources_string
+                 << "\". Valid"
+                 << " values are \"true\" or \"false\". Using default value of "
+                    "\"false\".";
+    }
+  }
+
+  if (!force_placed_resources) {
+    D3D12_FEATURE_DATA_D3D12_OPTIONS options = {};
+    if (SUCCEEDED(device->CheckFeatureSupport(D3D12_FEATURE_D3D12_OPTIONS,
+                                              &options, sizeof(options)))) {
+      return options.TiledResourcesTier >= D3D12_TILED_RESOURCES_TIER_1;
+    }
+  }
+
+  return false;
+}
+
+static uint64_t GetMaxHeapSizeInTiles() {
+  int64 override_value = 0;
+  Status s =
+      ReadInt64FromEnvVar("TF_DIRECTML_MAX_HEAP_TILES", 0, &override_value);
```

Comment:
This is more of a general comment, but we might consider putting the environment variables specific to tfdml in the README somewhere.

Reply:
https://github.com/microsoft/tensorflow-directml/wiki/Environment-Variables